### PR TITLE
More Markdown linting

### PR DESCRIPTION
**Pull request by plaindocs**
1. Switch all rules off. 
1. Add the the relevant ones back.
1. Automatically fix applicable rules.

Example config https://github.com/DavidAnson/markdownlint/blob/main/schema/.markdownlint.yaml#L3

- [x] Add to CI

Internal: DOC-160

### CODE REVIEW

**Comment by karensawrey** [comment]
This thing absolutely scares me in terms of home many tiny things it might break of not reviewed carefully. Will review it carefully first thing in the morning.

**Comment by plaindocs** [comment]
Agreed. If you can review tomorrow, we'll merge it, and then go slower in adding further checks every so often.

**Comment by karensawrey** [comment]
I'm replacing the `help` files messed up by linter from main for this PR. Let's exclude `pages/agent/v3/help/_annotate.md` from the list of files available for linting before we run the linter again. FAO @plaindocs

**Comment by karensawrey** [comment]
@keithduncan please take a look at the lists in https://github.com/buildkite/docs/pull/1231/files#diff-905a2cca8e473097eeba36d22883999120386ff427a8d7126710ac76beae521a - I've refactored the content and the levels a bit, I hope I didn't twist the meaning. 
Preview: https://docs-review-pr-1231.herokuapp.com/docs/agent/v3/aws/secrets-manager

File: pages/agent/v3/aws/secrets_manager.md [hunk 1]
```diff
@@ -22,13 +22,10 @@ for more details.
...
-	- The CloudFormation template includes an IAM policy with `kms:Decrypt` permission for this key.
+- Provide the Key ID (not the alias) used to encrypt the Secrets Manager secret to the `BuildkiteAgentTokenParameterStoreKMSKey` parameter. An IAM policy with `kms:Decrypt` permission for this key is included in the CloudFormation template.
 - Use the Secret Manager secret’s resource policy to grant `secretsmanager:GetSecretValue` permission to both the instance IAM role and the scaling Lambda IAM Role.
-  - Use your CloudFormation template’s Resources tab to find the `AutoscalingLambdaExecutionRole` and `IAMRole` roles, use their ARNs in the policy given below.
-	- Secret Manager will capture the role’s Unique ID when saving the resource
-  policy, if you re-create the IAM role you will need to save the resource
-  policy again to grant access.
+- Use your CloudFormation template’s Resources tab to find the `AutoscalingLambdaExecutionRole` and `IAMRole` roles, use their ARNs in the policy below.
+- Secret Manager will capture the role’s Unique ID when saving the resource policy; if you re-create the IAM role you will need to save the resource policy again to grant access.
```
Comment (keithduncan) [inline pages/agent/v3/aws/secrets_manager.md:28]: I would combine this dot point with the one above 😄

**Comment by keithduncan** [comment]
> @keithduncan please take a look at the lists in

Done, I made a tweak to the secrets manager docs but the list change looks good 😄

**Review by karensawrey — APPROVED** [review]
Many typos in small inconsistencies were fixed during the review. This is now good to hit prod.

**Pull request merged**